.buildkite/steps/build_and_push.sh

```diff
@@ -5,24 +5,18 @@ set -euo pipefail
 echo --- :hammer: Installing tools
 apk add helm yq skopeo git --quiet --no-progress
 
-tag=$(git describe)
-version=$(echo "$tag" | sed 's/v//')
-temp_agent_image=$(buildkite-agent meta-data get "agent-image")
-agent_image="${KO_DOCKER_REPO}/agent-k8s:${tag}"
-controller_image=$(buildkite-agent meta-data get "controller-image")
+. .buildkite/steps/repo_info.sh
```

Comment:
what does the preceding `.` do here? is it a fancy way of sourcing?

Reply:
I think it's the original way of sourcing, `source` being a synonym for `.` is a later development of some shells (probably even prior to BASH) and not part of POSIX.

See https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html where is says the result of the command `source` is unspecified. Whereas `.` is a [builtin](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#dot)